### DISCUSSION TITLE
Is the Excalidraw file format readable in other drawing apps?

### DISCUSSION THREAD

**Discussion by miguelmarcos** [excalidraw/excalidraw#3545] [General]
Hi. First off, congratulations and many thanks for such an awesome tool. Is the file readable in any existing diagram or drawing app? I know Excalidraw can export to PNG and SVG.

**Comment by dwelle** [comment]
Hey. Excalidraw uses a proprietary JSON schema to define the scene elements and state.

As of right now, only apps that directly account for Excalidraw can read excalidraw files. These are mostly other Excaldraw apps or apps that integrate [@excalidraw/excalidraw](https://www.npmjs.com/package/@excalidraw/excalidraw). For example, you can import exalidraw files to [Replit](http://replit.com/) (even though Replit modifies the editor and only allows a subset of tools, it can still read & render the full format — though you'll need to rename the file to `.draw` before import). Other apps, such as our own at [https://plus.excalidraw.com](plus.excalidraw.com), can read/write `.excalidraw` files directly.

As for inter-format converter, apart from the excalidraw → SVG, there's some being done on [SVG → excalidraw](https://github.com/excalidraw/svg-to-excalidraw).

What exactly were you interested in?

**Reply by miguelmarcos** [reply]
That's fair. I was interested in the ability to double-click on the Excalidraw file and open up Visio, for example, or whatever app I have the extension mapped to. I like that Obsidian hands off files to corresponding apps. I am all for interoperability. Nevertheless, Excalidraw is excellent and the existing functionality is a marvelous.

**Comment by maphew** [comment]
I'm interested in the inverse of this, what format(s) do I need to save-as from other prorgrams to import into Excalidraw?

**Reply by Mrazator** [reply]
ATM there is no out-of-the-box converter, so best you could do is likely SVG. There were some attempts to convert SVG into excalidraw (i.e. https://github.com/excalidraw/svg-to-excalidraw), but AFAIK they were far form perfect.
